js/data.js

```diff
@@ -6,15 +6,17 @@ var data = {
   editing: null,
   nextEntryId: 1
 };
-var entres = [];
+
 var previousEntriesJSON = localStorage.getItem('javascript-local-storage');
-function JsonSerialize() {
-  var entresJSON = JSON.stringify(entres);
-  localStorage.setItem('javascript-local-storage', entresJSON);
+function JsonSerialize(entry) {
+  var dataJSON = JSON.stringify(data.entries);
+  var currentView = data.view;
+  localStorage.setItem('javascript-local-storage', dataJSON);
+  localStorage.setItem('current screen', currentView);
```

Comment:
You should be saving the whole `data` object into localStorage, not just the entries & current view